x-pack/plugins/canvas/canvas_plugin_src/renderers/embeddable/embeddable_input_to_expression.ts

```diff
@@ -21,17 +21,24 @@ export const inputToExpressionTypeMap = {
 /*
   Take the input from an embeddable and the type of embeddable and convert it into an expression
 */
-export function embeddableInputToExpression(
-  input: EmbeddableInput,
+export function embeddableInputToExpression<
+  B extends boolean,
+  C = B extends true ? string : string | undefined
+>(
```

Comment:
This might be an over-complicated type - but basically, to avoid `ts-ignore` comments, I wanted to ensure that **when `useGenericEmbeddable` is true**, Typescript knows that the function will always return a string and **not** undefined.